rotkehlchen/data_import/importers/kucoin.py

```diff
@@ -0,0 +1,109 @@
+import csv
+from enum import Enum, auto
+from pathlib import Path
+from typing import Any
+
+from rotkehlchen.assets.converters import asset_from_kucoin
+from rotkehlchen.data_import.utils import BaseExchangeImporter
+from rotkehlchen.db.drivers.gevent import DBCursor
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.exchanges.data_structures import Trade
+from rotkehlchen.exchanges.utils import get_key_if_has_val
+from rotkehlchen.history.deserialization import deserialize_price
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_fee,
+    deserialize_timestamp_from_date,
+)
+from rotkehlchen.types import Location, TradeType
+
+
+class KucoinFileType(Enum):
+    """The type of the file we are parsing"""
+    TRADES = auto()  # default header format
+    TRADES_OLD = auto()  # old header format
+
+
+class KucoinImporter(BaseExchangeImporter):
+    @staticmethod
+    def _consume_trades(
+            csv_row: dict[str, str],
+            file_type: KucoinFileType,
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S',
+    ) -> Trade:
+        """
+        Consume entries of kucoin order history csv
+
+        May raise:
+        - KeyError
+        - DeserializationError
+        - UnknownAsset
+        """
+        try:
+            if file_type == KucoinFileType.TRADES:
```

Comment:
Why have another function that checks the same if again here? Just make two functions. One for importing old trades, one for importing new ones, with the common functonality abstracted and call them where you check the header